### PR TITLE
add blog for ingress2gateway v1.0

### PR DESCRIPTION
**Pull request by Stevenjin8**
### Description

Blog for the [Ingress2Gateway](https://github.com/kubernetes-sigs/ingress2gateway) v1.0 release.
Waiting for Gateway API v1.5 release and #54646.

### CODE REVIEW

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+
+You can also download the binary from [GitHub](https://github.com/kubernetes-sigs/ingress2gateway/releases/tag/v1.0.0) or [build from source](https://github.com/kubernetes-sigs/ingress2gateway/)
+
+### 2. Run Ingress2Gateway
+
+You can either pass Ingress2Gateway Ingress manifests, point it at a namespace in your cluster, or your whole cluster
+
+```bash
+# Pass it files
+i2gw print --input-file my-manifest.yaml,my-other-manifest.yaml --providers=ingress-nginx > gwapi.yaml
```
Comment (adrianmoisey) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: If I follow the instructions, the CLI name is incorrect:
```console
adrian@Adrians-MacBook-Pro:~$ brew install ingress2gateway
Warning: ingress2gateway 0.5.0 is already installed and up-to-date.
To reinstall 0.5.0, run:
  brew reinstall ingress2gateway
adrian@Adrians-MacBook-Pro:~$ i2gw
-bash: i2gw: command not found
adrian@Adrians-MacBook-Pro:~$ ingress2gateway
Convert Ingress manifests to Gateway API manifests

Usage:
  ingress2gateway [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command
  print       Prints Gateway API objects generated from ingress and provider-specific resources.
  version     Print the version number of ingress2gateway

Flags:
  -h, --help                help for ingress2gateway
      --kubeconfig string   The kubeconfig file to use when talking to the cluster. If the flag is not set, a set of standard locations can be searched for an existing kubeconfig file.

Use "ingress2gateway [command] --help" for more information about a command.
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+**I'm still waiting for the gateway api 1.5 release to update links**
+
+## Introduction
+
+[With the Ingress-NGINX retirement scheduled March 2026](/blog/2025/11/11/ingress-nginx-retirement/), the Kubernetes networking landscape is at a turning point.
+For most organizations, the question isn't whether to migrate to Gateway API, but how to do so safely.
+
+Migrating from Ingress to Gateway API is a fundamental shift in API design.
+Gateway API provides a modular, extensible, tenancy-aware design.
```
Comment (danwinship) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: readers who aren't familiar with Gateway API may not know what you mean by "tenancy-aware"

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+## Introduction
+
+[With the Ingress-NGINX retirement scheduled March 2026](/blog/2025/11/11/ingress-nginx-retirement/), the Kubernetes networking landscape is at a turning point.
+For most organizations, the question isn't whether to migrate to Gateway API, but how to do so safely.
+
+Migrating from Ingress to Gateway API is a fundamental shift in API design.
+Gateway API provides a modular, extensible, tenancy-aware design. 
+Conversely, the Ingress API is overly simple, and implementations such as Ingress NGINX extend the API through esoteric annotations, configmaps, and CRDs.
+This flat, annotation-heavy model leads to few tenancy controls and lots of implementation-specific behavior.
```
Comment (danwinship) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: The "annotation-heavy"/"implementation-specific" thing and the "few tenancy controls" thing are independent problems, and it might be better to explain them separately (if briefly).

ie, the two things that make migration hard are: (1) even where the features are similar, the configuration is very different, (2) Gateway splits things up into more objects than Ingress, to better support clusters where different users have different levels of privilege.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+Migrating away from Ingress controllers such as Ingress NGINX presents the daunting task of capturing all the nuances of the Ingress controller, and mapping that behavior to Gateway API and implementation-specific Gateway API extensions.
+
+Ingress2Gateway is an assistant that helps teams confidently move from Ingress to Gateway API.
+It translates Ingress resources and implementation-specific annotations to Gateway API while warning you about untranslatable configuration and offering suggestions.
+
+Today, SIG Network is proud to announce the **1.0 release of Ingress2Gateway**.
+This milestone represents a comprehensive, stable, tested migration assistant for teams ready to modernize their networking stack.
```
Comment (danwinship) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: "comprehensive" is a bit of a stretch isn't it?

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+### Ingress NGINX Annotation Support
+
+The main improvement for the 1.0 release is more comprehensive Ingress NGINX support.
+Before the 1.0 release, Ingress2Gateway only supported three Ingress NGINX annotations.
+For the 1.0 release, Ingress2Gateway supports over 30 common annotations (CORS, Backend TLS, regex matching, path rewrite, etc.).
+
+### E2E Tests for Ingress NGINX
+
+Each supported Ingress NGINX annotation, and representative combinations of common annotations, is backed by controller-level integration tests that verify the behavioral equivalence of the Ingress NGINX configuration and the generated Gateway API.
+These tests exercise real controllers in live clusters and compare runtime behavior (routing, redirects, rewrites, etc.), not just YAML structure.
```
Comment (danwinship) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md:36]: (oh wow, this is great)

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+          ingress-nginx only supports TCP-level timeouts.
+          i2gw performed a best-effort translation to
+          timeouts.request.
+          Please verify this meets your needs:
+          https://gateway-api.sigs.k8s.io/guides/http-timeouts/
+
+[WARNING] HTTPRoute my-api/my-api-my-api-example-com
+          ingress-nginx regex matches are case-insensitive prefix matches.
+          You might want to change this.
+```
```
Comment (danwinship) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md:258]: It might be good to have the example include some obscure annotation that i2gw doesn't recognize at all, to show what it does in that case?

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+
+Ingress2Gateway made a best-effort translation from the `nginx.ingress.kubernetes.io/proxy-{send,read}-timeout` annotations to a 10 second [request timeout](https://gateway-api.sigs.k8s.io/guides/http-timeouts/) in our HTTP route.
+If requests for this service should be much shorter, say 3 seconds, you can make the corresponding changes to our Gateway API manifests.
```
Comment (danwinship) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
If requests for this service should be much shorter, say 3 seconds, you can make the corresponding changes to your Gateway API manifests.
```
?

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 8]
```diff
@@ -0,0 +1,348 @@
+---
+title: "Announcing Ingress2Gateway 1.0: Your Path to Gateway API"
+slug: ingress2gateway-1-0-release
+tags: ["Gateway API", "Networking", "SIG Network", "Migration"]
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
```
We don't use tags; the theme supports them, but I wouldn't add them without a discussion period.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+authors: [
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: It's actually `author`, free text (but usually a comma separated list).

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 10]
```diff
@@ -0,0 +1,348 @@
...
+    "[Beka Modebadze](https://github.com/bexxmodd) (Google)",
+    "[Steven Jin](https://github.com/Stevenjin8) (Microsoft)"
+]
+draft: true
+---
+
+**I'm still waiting for the gateway api 1.5 release to update links**
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
```
(before merge)

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+
+## Introduction
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
```

No need; the introductory nature is implied. And we very commonly don't start our articles with a level 2 heading.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 12]
```diff
@@ -0,0 +1,348 @@
...
+
+[With the Ingress-NGINX retirement scheduled March 2026](/blog/2025/11/11/ingress-nginx-retirement/), the Kubernetes networking landscape is at a turning point.
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: (nit) Try not to make a long phrase be the anchor test. Maybe just "retirement".

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 13]
```diff
@@ -0,0 +1,348 @@
...
+For most organizations, the question isn't whether to migrate to Gateway API, but how to do so safely.
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Make "Gateway API" a hyperlink.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 14]
```diff
@@ -0,0 +1,348 @@
...
+
+Migrating from Ingress to Gateway API is a fundamental shift in API design.
+Gateway API provides a modular, extensible, tenancy-aware design. 
+Conversely, the Ingress API is overly simple, and implementations such as Ingress NGINX extend the API through esoteric annotations, configmaps, and CRDs.
+This flat, annotation-heavy model leads to few tenancy controls and lots of implementation-specific behavior.
+Migrating away from Ingress controllers such as Ingress NGINX presents the daunting task of capturing all the nuances of the Ingress controller, and mapping that behavior to Gateway API and implementation-specific Gateway API extensions.
+
+Ingress2Gateway is an assistant that helps teams confidently move from Ingress to Gateway API.
+It translates Ingress resources and implementation-specific annotations to Gateway API while warning you about untranslatable configuration and offering suggestions.
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Does it work on resources, manifests, or both?

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 15]
```diff
@@ -0,0 +1,348 @@
...
+
+Today, SIG Network is proud to announce the **1.0 release of Ingress2Gateway**.
+This milestone represents a comprehensive, stable, tested migration assistant for teams ready to modernize their networking stack.
+
+## Ingress2Gateway 1.0
+
+### Ingress NGINX Annotation Support
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
### Ingress NGINX annotation support
```
Following the house style adds gravitas.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 16]
```diff
@@ -0,0 +1,348 @@
...
+
+The main improvement for the 1.0 release is more comprehensive Ingress NGINX support.
+Before the 1.0 release, Ingress2Gateway only supported three Ingress NGINX annotations.
+For the 1.0 release, Ingress2Gateway supports over 30 common annotations (CORS, Backend TLS, regex matching, path rewrite, etc.).
+
+### E2E Tests for Ingress NGINX
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Following the house style adds gravitas.

Try:
```suggestion
### Comprehensive integration testing
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 17]
```diff
@@ -0,0 +1,348 @@
...
+
+* spin up an Ingress NGINX controller
+* spin up multiple Gateway API controllers
+* apply Ingress resources that have implementation-specific configuration
+* translate Ingress resources to Gateway API with `ingress2gateway` and apply generated resources
+* verify that the Gateway API controllers and the Ingress controller exhibit equivalent behavior.
+
+A comprehensive test suite not only catches bugs in development, but also ensures the correctness of the translation, especially given [surprising edge cases and unexpected defaults](Link to article with these), so that you don't find out about them in production.
+
+### Notification & Error Handling System
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Write this in sentence case (to match our house style). The more important the article, the more we want people to feel that the message has authority and has been checked through.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 18]
```diff
@@ -0,0 +1,348 @@
...
+Ingress2Gateway is a migration assistant, not a one-shot replacement.
+Its goal is to
+
+* migrate supported Ingress configuration and behavior
+* identify unsupported configuration and suggest alternatives
+* reevaluate and potentially discard undesirable configuration
+
+The rest of the section shows you how to safely migrate the following Ingress NGINX configuration
+
+```yaml=
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Drop the trailing = unless you have evidence it won't be a maintenance headache. It safe to omit.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 19]
```diff
@@ -0,0 +1,348 @@
...
+        - my-api.example.com
+      secretName: my-api-tls
+```
+
+### 1. Install the Toolkit
+
+Install the binary with
+
+```bash
+go install github.com/kubernetes-sigs/ingress2gateway@v1.0.0
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md:104]: Is this the official way to install it?

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 20]
```diff
@@ -0,0 +1,348 @@
...
+                  number: 80
+            path: /users/(\d+)
+            pathType: ImplementationSpecific
+  tls:
+    - hosts:
+        - my-api.example.com
+      secretName: my-api-tls
+```
+
+### 1. Install the Toolkit
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Write this in sentence case (to match our house style). The more important the article, the more we want people to feel that the message has authority and has been checked through.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 21]
```diff
@@ -0,0 +1,348 @@
...
+Ingress2Gateway made a best-effort translation from the `nginx.ingress.kubernetes.io/proxy-{send,read}-timeout` annotations to a 10 second [request timeout](https://gateway-api.sigs.k8s.io/guides/http-timeouts/) in our HTTP route.
+If requests for this service should be much shorter, say 3 seconds, you can make the corresponding changes to our Gateway API manifests.
+
+To match Ingress NGINX default behavior, Ingress2Gateway also added a listener on port 80 and a [HTTP Request redirect filter](https://gateway-api.sigs.k8s.io/reference/spec/#httprequestredirectfilter) to redirect HTTP traffic to HTTPS.
+Some organizations might not want to serve HTTP traffic at all and remove the listener on port 80 and the corresponding HTTPRoute.
+
+There is also a warning that Ingress NGINX regex matches are case-insensitive prefix matches, which is why there is a match pattern of `(?i)/users/(\d+).*`.
+Most organizations will want to change this behavior to be an exact case-sensitive match by remove the leading `(?i)` and the trailing `.*` from the path pattern.
+
+**WARNING:** Always thoroughly review the generated output and logs.
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Use a Docsy caution callout if you can (see website style guide). This doesn't quite merit a warning.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 22]
```diff
@@ -0,0 +1,348 @@
...
+This way, you can quickly recover from any misconfiguration that made it through your tests.
+
+Finally, when you have shifted all your traffic to your Gateway API controller, delete your Ingress resources and your Ingress NGINX controller.
+
+## Conclusion
+
+The Ingress2Gateway 1.0 release is just the beginning, and we hope that you use Ingress2Gateway to safely migrate to Gateway API.
+As we approach the March 2026 Ingress-NGINX retirement, we invite the community to help us increase our configuration coverage, expand testing, and improve UX.
+
+## Gateway API Resources
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Write this in sentence case (to match our house style). The more important the article, the more we want people to feel that the message has authority and has been checked through.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 23]
```diff
@@ -0,0 +1,348 @@
+---
+title: "Announcing Ingress2Gateway 1.0: Your Path to Gateway API"
+slug: ingress2gateway-1-0-release
+tags: ["Gateway API", "Networking", "SIG Network", "Migration"]
+authors: [
+    "[Beka Modebadze](https://github.com/bexxmodd) (Google)",
+    "[Steven Jin](https://github.com/Stevenjin8) (Microsoft)"
+]
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Try this:
```suggestion
author: >
  [Beka Modebadze](https://github.com/bexxmodd) (Google),
  [Steven Jin](https://github.com/Stevenjin8) (Microsoft)
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 17 above)
Comment (lmktfy) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
### Notifications & error handling
```
?

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 24]
```diff
@@ -0,0 +1,348 @@
...
+* reevaluate and potentially discard undesirable configuration
+
+The rest of the section shows you how to safely migrate the following Ingress NGINX configuration
+
+```yaml=
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    nginx.ingress.kubernetes.io/proxy-body-size: "1g"
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: If this also works, let's use uppercase:
```suggestion
    nginx.ingress.kubernetes.io/proxy-body-size: "1G"
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 20 above)
Comment (lmktfy) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
### 1. Install the toolkit
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 25]
```diff
@@ -0,0 +1,348 @@
...
+go install github.com/kubernetes-sigs/ingress2gateway@v1.0.0
+```
+
+or
+
+```bash
+brew install ingress2gateway
+```
+
+You can also download the binary from [GitHub](https://github.com/kubernetes-sigs/ingress2gateway/releases/tag/v1.0.0) or [build from source](https://github.com/kubernetes-sigs/ingress2gateway/)
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: I don't know if we can make it available via https://dl.k8s.io/ but it would be nice if we could try.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 26]
```diff
@@ -0,0 +1,348 @@
...
+# Or your whole cluster
+i2gw print --providers=ingress-nginx > gwapi.yaml
+```
+
+**Note:** You can also pass `--emitter <kgateway|envoy-gateway>` to output implementation-specific extensions. 
+
+### 3. Review the output
+
+This is the most critical step.
+Ingress2Gateway outputs Gateway API resources to `gwapi.yaml`, and it also emits warnings that explain what did not translate exactly and what to review manually.
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
Ingress2Gateway writes a Gateway API manifest to `gwapi.yaml`, and it also emits warnings
that explain what did not translate exactly and what to review manually.
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 27]
```diff
@@ -0,0 +1,348 @@
...
+
+```yaml=
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Drop the trailing = unless you have evidence it won't be a maintenance headache. It safe to omit.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 28]
```diff
@@ -0,0 +1,348 @@
...
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  annotations:
+    gateway.networking.k8s.io/generator: ingress2gateway-dev
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md:142]: I'll file an issue that this is unregistered.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 14 above)
Comment (Stevenjin8) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: by resources and manifests, do you mean can it read from a kubeapi server and/or a bunch of yaml files? If so, the answer is yes.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 26 above)
Comment (Stevenjin8) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: hmm I dont think either is quite right. I'll just say "the commands from the previous sections"

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 19 above)
Comment (Stevenjin8) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md:104]: yes

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 3 above)
Comment (Stevenjin8) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: I might just skip the tenancy point. We're not trying to sell gateway api and although the tenancy story of Ingress is complicated, it doesn't pose a migration challenge. Its quite easy to use Gateway API in a single tenant environemnt.

**Comment by lmktfy** [comment]
I expect that we can merge this as a draft once it's almost ready. A follow up PR can confirm the release date and hyperlinks to the new tool.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md (hunk 28 above)
Comment (lmktfy) [reply content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md:142]: (issue filed)

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 29]
```diff
@@ -0,0 +1,355 @@
...
+  parents: null
+```
+
+Ingress2Gateway successfully translated some annotations into their Gateway API equivalents.
+For example, the `nginx.ingress.kubernetes.io/enable-cors` annotation was translated into a CORS filter.
+But upon closer inspection, the `nginx.ingress.kubernetes.io/proxy-{read,send}-timeout` and `nginx.ingress.kubernetes.io/proxy-body-size` annotations do not map perfectly.
+The logs show the reason for these omissions as well as reasoning behind the translation
+
+```
+# TODO we still need to figure out what these look like in ingress2gateway logs, but the content should be similar to the following
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Marking this TODO with a comment.

**Comment by lmktfy** [comment]
BTW, the [Netlify preview ](https://deploy-preview-54672--kubernetes-io-main-staging.netlify.app/blog/1/01/01/ingress2gateway-1-0-release/)is available.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 30]
```diff
@@ -0,0 +1,355 @@
...
+            pathType: ImplementationSpecific
+  tls:
+    - hosts:
+        - my-api.example.com
+      secretName: my-api-tls
+```
+
+### 1. Install Ingress2Gateway
+
+Install the binary with
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Maybe something like this:
```suggestion 
Set up a Go development environment if you don't already have one.
Then, install the binary by running:
```
?

I think it's likely that most readers won't be able to run `go install <foo>` successfully. But on the other hand, they can probably figure it out given half an hour.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 31]
```diff
@@ -0,0 +1,355 @@
...
+---
+
+**I'm still waiting for the gateway api 1.5 release to update links**
+
+With the Ingress-NGINX [retirement](/blog/2025/11/11/ingress-nginx-retirement/) scheduled for March 2026, the Kubernetes networking landscape is at a turning point.
+For most organizations, the question isn't whether to migrate to [Gateway API](https://gateway-api.sigs.k8s.io/), but how to do so safely.
+
+Migrating from Ingress to Gateway API is a fundamental shift in API design.
+Gateway API provides a modular, extensible API with strong support for Kubernetes-native RBAC.
+Conversely, the Ingress API is overly simple, and implementations such as Ingress-NGINX extend the API through esoteric annotations, configmaps, and CRDs.
```
Comment (bexxmodd) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
Conversely, the Ingress API is overly simple, and implementations such as Ingress-NGINX extend the API through esoteric annotations, ConfigMaps, and CRDs.
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 32]
```diff
@@ -0,0 +1,355 @@
...
+# Or your whole cluster
+ingress2gateway print --providers=ingress-nginx > gwapi.yaml
+```
+
+**Note:** You can also pass `--emitter <kgateway|envoy-gateway>` to output implementation-specific extensions. 
+
+### 3. Review the output
+
+This is the most critical step.
+The commands from the previous section output Gateway API manifest to `gwapi.yaml`, and they also emits warnings that explain what did not translate exactly and what to review manually.
```
Comment (bexxmodd) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
The commands from the previous section output Gateway API manifests to `gwapi.yaml`, and they also emit warnings that explain what did not translate exactly and what to review manually.
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 33]
```diff
@@ -0,0 +1,355 @@
...
+        statusCode: 308
+      type: RequestRedirect
+status:
+  parents: null
+```
+
+Ingress2Gateway successfully translated some annotations into their Gateway API equivalents.
+For example, the `nginx.ingress.kubernetes.io/enable-cors` annotation was translated into a CORS filter.
+But upon closer inspection, the `nginx.ingress.kubernetes.io/proxy-{read,send}-timeout` and `nginx.ingress.kubernetes.io/proxy-body-size` annotations do not map perfectly.
+The logs show the reason for these omissions as well as reasoning behind the translation
```
Comment (bexxmodd) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
The logs show the reason for these omissions as well as reasoning behind the translation.
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 34]
```diff
@@ -0,0 +1,355 @@
...
+        value: (?i)/users/(\d+).*
+    name: rule-0
+    timeouts:
+      request: 3s
+```
+
+### 4. Verify
+
+Now that you have Gateway API manifests, you should thoroughly test them in a development cluster.
+In this case, you should at least double-check that your Gateway API implementation's maximum body size defaults are appropriate for you and verify that a three second timeout is enough.
```
Comment (bexxmodd) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
In this case, you should at least double-check that your Gateway API implementation's maximum body size defaults are appropriate for you and verify that a three-second timeout is enough.
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 35]
```diff
@@ -0,0 +1,355 @@
...
+
+```bash
+brew install ingress2gateway
+```
+
+You can also download the binary from [GitHub](https://github.com/kubernetes-sigs/ingress2gateway/releases/tag/v1.0.0) or [build from source](https://github.com/kubernetes-sigs/ingress2gateway/)
+
+### 2. Run Ingress2Gateway
+
+You can either pass Ingress2Gateway Ingress manifests, point it at a namespace in your cluster, or your whole cluster
```
Comment (bexxmodd) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]:
```suggestion
You can pass Ingress2Gateway Ingress manifests, or point the tool at a specific namespace or your entire cluster.
```

**Comment by lmktfy** [comment]
BTW, when we publish this, we could update https://kubernetes.io/blog/2023/10/25/introducing-ingress2gateway/ to mention the newer blog article.

Not only we could, I recommend we do.

**Comment by Stevenjin8** [comment]
@lmktfy @bexxmodd please take another look. Hoping to release this maybe 3/20?

File: content/en/blog/_posts/2023/introducing-ingress2gateway/index.md
Comment (lmktfy) [inline content/en/blog/_posts/2023/introducing-ingress2gateway/index.md:1]: ## :information_source: Important!

The changes to this file need to not be part of _this_ PR. Save them for a separate PR that we can hold, approve, and then unhold once the release has happened.

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 36]
```diff
@@ -0,0 +1,362 @@
...
+The main improvement for the 1.0 release is more comprehensive Ingress-NGINX support.
+Before the 1.0 release, Ingress2Gateway only supported three Ingress-NGINX annotations.
+For the 1.0 release, Ingress2Gateway supports over 30 common annotations (CORS, backend TLS, regex matching, path rewrite, etc.).
+
+### Comprehensive integration testing
+
+Each supported Ingress-NGINX annotation, and representative combinations of common annotations, is backed by controller-level integration tests that verify the behavioral equivalence of the Ingress-NGINX configuration and the generated Gateway API.
+These tests exercise real controllers in live clusters and compare runtime behavior (routing, redirects, rewrites, etc.), not just YAML structure.
+
+Our tests:
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: ("important nit")
```suggestion
The tests:
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 37]
```diff
@@ -0,0 +1,362 @@
...
+```bash
+# Pass it files
+ingress2gateway print --input-file my-manifest.yaml,my-other-manifest.yaml --providers=ingress-nginx > gwapi.yaml
+# Use a namespace in your cluster
+ingress2gateway print --namespace my-api --providers=ingress-nginx > gwapi.yaml
+# Or your whole cluster
+ingress2gateway print --providers=ingress-nginx > gwapi.yaml
+```
+
+**Note:** You can also pass `--emitter <kgateway|envoy-gateway>` to output implementation-specific extensions.
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: Given we use another callout later, how about this:
```suggestion
{{< note >}}
You can also pass `--emitter <kgateway|envoy-gateway>` to output implementation-specific extensions.
{{< /note >}}
```

File: content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md [hunk 38]
```diff
@@ -0,0 +1,362 @@
...
+This way, you can quickly recover from any misconfiguration that made it through your tests.
+
+Finally, when you have shifted all your traffic to your Gateway API controller, delete your Ingress resources and uninstall Ingress-NGINX.
+
+## Conclusion
+
+The Ingress2Gateway 1.0 release is just the beginning, and we hope that you use Ingress2Gateway to safely migrate to Gateway API.
+As we approach the March 2026 Ingress-NGINX retirement, we invite the community to help us increase our configuration coverage, expand testing, and improve UX.
+
+## Gateway API resources
```
Comment (lmktfy) [inline content/en/blog/_posts/2026/ingress2gateway-v1-0-release.md]: (nit)
```suggestion
## Resources about Gateway API
```

This new wording is different from the HTTP sense of "resource", which comes up a lot.
For example, `/apis/gateway.networking.k8s.io/v1/namespaces/default/gateways/` is a resource (it's a collection of Gateways).

File: content/en/blog/_posts/2023/introducing-ingress2gateway/index.md [hunk 39]
```diff
@@ -8,6 +8,9 @@ author: >
   Kobi Levi (independent)
 ---
 
+_**Update**: [Ingress2Gateway v1.0](/blog/2026/ingress2gateway-v1-0-release/) is
```
Comment (lmktfy) [inline content/en/blog/_posts/2023/introducing-ingress2gateway/index.md]: This URL isn't correct; the real URL will have a month and date in its path.

You can use an [alias](https://gohugo.io/methods/page/aliases/) to avoid that (but it still needs to be in a separate PR, so I wouldn't bother with `aliases`).
